File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
int tribo[38] = { -1 };
class Solution {
public:
    int tribonacci(int n) {
        if (tribo[0] == -1) {
            tribo[0] = 0;
            tribo[1] = tribo[2] = 1;
            for (int i = 3; i <= 37; i++) {
                tribo[i] = tribo[i - 1] + tribo[i - 2] + tribo[i - 3];
            }
        }
        return tribo[n];
    }
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        if (n <= 2) {
            return Math.min(n, 1);
        }
        int a = 0, b = 1, c = 1;
        for (int i = 3; i <= n; i++) {
            int d = a + b + c;
            a = b;
            b = c;
            c = d;
        }
        return c;
    }
}